docs/arch-split.md

```diff
@@ -408,23 +458,28 @@ We see no reason to redo that previous work, so the above still
 describes the current state of the abstract spec and some of the
 invariants.
 
+
 ### Theory-specific architecture-generic locales
 
 For further updates, however, we have developed a new pattern which we
 hope will eliminate the need for more "Pre" theories, and only require
 the addition of Arch theories for each existing theory.
 
-In this pattern, an existing theory Foo_AI is split into two theories:
+In this pattern, an existing theory `Foo_AI` is split into two theories:
 
-- `Foo_AI` retains the architecture-generic parts, using a locale `Foo_AI`
+- `Foo_AI` retains the architecture-generic parts, using locales `Foo_AI*`
    where necessary to *assume* the existence of the appropriate
    architecture-specific parts.
+   Note: if only one locale is needed, it is named `Foo_AI` by convention,
+   otherwise naming proceeds: `Foo_AI_1`, `Foo_AI_2`, etc.
```

Comment:
I wonder if the `_1` is going to bite us at some point. It *should* be fine, but if we end up getting theorems whose names end in `_1` etc, then Isabelle's internal indexing into fact lists might get confused (shouldn't, but from time to time some code resurfaces that relies on naming).

Anyway, nothing to do right now, just wanted to point out a potential future hazard. So far, in AInvs, this seems to have worked out fine.

Reply:
I don't think there will be a problem, they will be prefixed with the locale name so `whatever_AI_1.foo`. However if you want to change this, there are precisely three so far: `Deterministic_AI_1, Tcb_AI_1, Finalise_AI_1`, so it'll be easy.

Reply:
Good point. Happy to just leave it as is.